### ISSUE TITLE
User location and following buttons on map disappear when location services are turned off

### ISSUE THREAD

**Issue by achou11** [digidem/CoMapeo-mobile#143]
_Transferred from https://github.com/digidem/CoMapeo-mobile/pull/136#issuecomment-1865105011_

---

Just playing around with the location services behavior and noticing different behavior here than in Mapeo Mobile

Basic steps are:

1. Open app
2. Turn off location services
3. Return to app and wait for a few seconds
4. Turn on location services
5. Return to app and wait for a few seconds

---

- Mapeo:

https://github.com/digidem/CoMapeo-mobile/assets/18542095/ffaa2e95-91e8-44b3-aa09-40340c683db3

- CoMapeo:

https://github.com/digidem/CoMapeo-mobile/assets/18542095/f3146222-a18d-4cd9-b73c-f6c02bf82bcd

You'll notice the main differences are the related to the user location marker and following button after step 3. In Mapeo, they remain on the screen. In CoMapeo, they disappear, but return once the location services are on.

What's happening in CoMapeo seems less desirable, but I don't actually know what the desired behavior is.

**Comment by achou11** [comment]
cc @gmaclennan maybe you have some additional context here?

**Comment by achou11** [comment]
From Erik, about why what Mapeo is doing doesn't seem like desired behavior (https://github.com/digidem/CoMapeo-mobile/pull/136#issuecomment-1865143719):

> My reasoning for why I don't think it is desired behaviour is for 2 reasons:
>
>  1. it could mislead the user into thinking that there location service is on/confuse them as to where they currently are.
>  2. Security reasons. If they turn off there location services at a sensitive site, it will mark there last location at that sensitive site. So if someone (eg security guard) was to check their phone, there would be a record that they were at that sensitive site.

I replied with the following:



>>    it could mislead the user into thinking that there location service is on/confuse them as to where they currently are.
>
> In Mapeo, there's the gps pill in the map screen UI, which I think is supposed to address this concern. Currently not implemented in CoMapeo
>
>>    Security reasons. If they turn off there location services at a sensitive site, it will mark there last location at that sensitive site. So if someone (eg security guard) was to check their phone, there would be a record that they were at that sensitive site.
>
> yeah this is a good point! i was coming from the angle that location services being available is not always due to an explicit action. For example if you have a hardware issue i think that could cause this behavior. In that case, not sure I'd want things that were showing up a few seconds ago to disappear

**Comment by achou11** [comment]
> i was coming from the angle that location services being available is not always due to an explicit action. For example if you have a hardware issue i think that could cause this behavior.
thinking about Erik's comments and combined with my uncertainties around this statement i made, I'm thinking that maybe what CoMapeo is doing is closer to the desired behavior. 

Think the issue I'm documenting is more about consistency with Mapeo and making sure that any notable deviations are justified.

**Comment by gmaclennan** [comment]
Yes I'm unsure about the desired behaviour, and I think we should take this question to the co-design team. From the technical side, I think what we are actually interested in is whether a location is "stale", e.g. it refers to a location from more than X milliseconds ago. For the user it would be the same whether it's because they have turned off location or because the weather has obscured GPS reception. I wrote a quick idea about how we could implement this https://github.com/digidem/CoMapeo-mobile/commit/7323764db81788f9eb774d991583fee0b1136c78. If co-design team agrees, we can hide the location marker on the map when the location is stale, and in a future implementation discuss whether there is a different way of continuing to display the user location, but with different symbology that indicates that the location is now "stale".
